lib/docker-toolbelt.coffee

```diff
@@ -359,7 +363,7 @@ DockerToolbelt::compileRegistryAndName = Promise.method ({ registry = '', imageN
 			tagName = 'latest'
 		return "#{registry}#{imageName}:#{tagName}"
 	else
-		return "#{registry}#{imageName}@#{digest}"
+		return "#{registry}#{imageName}#{if tagName then ':' + tagName else ''}@#{digest}"
```

Comment:
This is a different bug: the `tagName` was being discarded when a `digest` value was provided.

Reply:
This is definitely not a bug -- a digest uniquely identifies an image so tag is redundant. In fact, I think including both a tag and a digest might be invalid as per the various Docker specs (which I'd rather not have to go lookup right now :).

Reply:
@dfunckt is totally right, it's either or with these.

Reply:
Probably best to add a comment here to make the intent more clear.

Reply:
OK, I am reverting the change to this line, but... The `getRegistryAndName()` method is happy to parse a value like `name.com/repo/image:tag@digest` and save both the tag and digest in separate fields, and will not throw an `'Invalid image name'` error when both the tag and digest are provided. Then `compileRegistryAndName()` simply discards the tag (when a digest exists). If an image name with both a tag and a digest is parsed as a valid name, it seems a bit odd to discard the tag when reassembling the parts.